src/context/SessionContext.ts

```diff
@@ -0,0 +1,5 @@
+import { createContext } from "react";
+import { SessionContextType } from "../types/index";
+
+export const SessionContext =
+    createContext<SessionContextType | undefined>(undefined);
```

Comment:
I don't think you need to mention undefined implicity, if you write the below code , then it will infer undefined by default, let me know if this works or not

```suggestion
    createContext<SessionContextType |>();
```